src/gebt_poc/solver.cpp

```diff
@@ -557,4 +557,67 @@ Kokkos::View<double**> CalculateStaticIterationMatrix(
     return iteration_matrix;
 }
 
+Kokkos::View<double*> ConstraintsResidualVector(
+    const Kokkos::View<double*> gen_coords, const Kokkos::View<double*> position_vector
+) {
+    auto rotation_0 = gen_alpha_solver::EulerParameterToRotationMatrix(
+        gen_alpha_solver::create_vector({gen_coords(3), gen_coords(4), gen_coords(5), gen_coords(6)})
+    );
+    auto x0 =
+        gen_alpha_solver::create_vector({position_vector(0), position_vector(1), position_vector(2)}
+        );
+    auto u0 = gen_alpha_solver::create_vector({gen_coords(0), gen_coords(1), gen_coords(2)});
+    auto x0_u0 = Kokkos::View<double*>("x0_u0", kNumberOfVectorComponents);
+    Kokkos::parallel_for(
+        kNumberOfVectorComponents, KOKKOS_LAMBDA(const size_t i) { x0_u0(i) = x0(i) + u0(i); }
+    );
+    auto R_x0u0 = gen_alpha_solver::multiply_matrix_with_vector(rotation_0, x0_u0);
```

Comment:
Use `KokkosBlas` to multiply the rotation matrix by `x0_u0` and subtract `x0` using `gemv`. This avoids several allocations and two `parallel_for` calls. It also avoids the reimplementation of matrix-vector multiply. The result is stored directly in the `constraint_residual` since `R_x0u0_x0` is a subview.

```c++
    // Copy x0 into R_x0u0_x0 and calculate R*x0u0-x0 with gemv
    Kokkos::deep_copy(R_x0u0_x0, x0);
    KokkosBlas::gemv("N", 1.0, rotation_0, x0_u0, -1.0, R_x0u0_x0);
```